core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -460,9 +460,17 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       val newLastScanTime = clock.getTimeMillis()
       logDebug(s"Scanning $logDir with lastScanTime==$lastScanTime")
 
+      val notStale = mutable.HashSet[String]()
```

Comment:
it would be nice to add a comment here describing notStale and why we need it.

Reply:
@tgravescs thank you for the review!
PR has been updated with a comment.